datafusion/core/src/physical_plan/joins/hash_join.rs

```diff
@@ -1173,6 +1149,18 @@ impl HashJoinStream {
         };
         build_timer.done();
 
+        // Reserving memory for visited_left_side bitmap in case it hasn't been initialied yet
+        // and join_type requires to store it
+        if self.visited_left_side.is_none()
```

Comment:
You might be able to make this less verbose by moving the initialization into  the initialization of `visited_left_size`


```
        let visited_left_side = self.visited_left_side.get_or_insert_with(|| {
            let visited_bitmap_size = bit_util::ceil(left_data.1.num_rows(), 8);
            self.reservation.lock().try_grow(visited_bitmap_size)?;
            self.join_metrics.build_mem_used.add(visited_bitmap_size);
            ... 
```